### PR TITLE
Autosave plugin should ignore remote changes

### LINKED ISSUE #9233: Autosave plugin not ignoring `transparent` type events.
The current autosave calls the Promise for almost all `change:data` events. This is not needed as some changes may be from the Collaboration Mode, (ie, changes from other users). This will cause in collaboration mode:

- Unwanted API calls for every connected user.
- Unwanted API calls for other ignorable events.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@ckeditor/ckeditor5-autosave/src/autosave.js b/node_modules/@ckeditor/ckeditor5-autosave/src/autosave.js
index 004c0a7..20720ef 100644
--- a/node_modules/@ckeditor/ckeditor5-autosave/src/autosave.js
+++ b/node_modules/@ckeditor/ckeditor5-autosave/src/autosave.js
@@ -153,11 +153,17 @@ export default class Autosave extends Plugin {
 
 		this._pendingActions = editor.plugins.get( PendingActions );
 
-		this.listenTo( doc, 'change:data', () => {
+		this.listenTo( doc, 'change:data', (_e, batch) => {
 			if ( !this._saveCallbacks.length ) {
 				return;
 			}
 
+      // If the change is from another collaborative session from another user
+      // don't autosave.
+      if ( batch.type === 'transparent') {
+        return;
+      }
+
 			if ( this.state == 'synchronized' ) {
 				this._action = this._pendingActions.add( t( 'Saving changes' ) );
 				this.state = 'waiting';
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Feature (autosave): \`Autosave#save()\` will now return a promise that is resolved when the autosave callback has finished.

Other (autosave): Autosave plugin will now ignore changes coming from remote clients during real-time collaboration. Closes #9233.

---

### Additional information

Requires #11009.

### CODE REVIEW

File: packages/ckeditor5-autosave/src/autosave.js [hunk 1]
```diff
@@ -109,6 +110,17 @@ export default class Autosave extends Plugin {
 		 */
 		this._lastDocumentVersion = editor.model.document.version;
 
+		/**
+		 * Promised used for asynchronous save calls.
```
Comment (niegowski) [inline packages/ckeditor5-autosave/src/autosave.js:114]:
```suggestion
		 * Promise used for asynchronous save calls.
```

File: packages/ckeditor5-autosave/src/autosave.js [hunk 2]
```diff
@@ -224,12 +240,19 @@ export default class Autosave extends Plugin {
 	 * @private
```
Comment (niegowski) [inline packages/ckeditor5-autosave/src/autosave.js:240]: Missing return docs.

File: packages/ckeditor5-autosave/src/autosave.js [hunk 3]
```diff
@@ -224,12 +240,19 @@ export default class Autosave extends Plugin {
...
 	 */
 	_save() {
+		if ( this._savePromise ) {
+			return this._savePromise;
```
Comment (niegowski) [inline packages/ckeditor5-autosave/src/autosave.js:244]: Let's make sure that the returned promise is resolved only after the data is truly saved (the one that was in the editor while the `save()` method was called.

**Review by niegowski — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-autosave/src/autosave.js [hunk 4]
```diff
@@ -222,39 +247,91 @@ export default class Autosave extends Plugin {
...
 	 *
 	 * @private
+	 * @param {Boolean} [manualSave=false] Whether the save was triggered by another plugin (`true`) or by the autosave plugin, after
+	 * the model has changed (`false`).
+	 * @returns {Promise} A promise that will be resolved when the autosave callback is finished.
 	 */
-	_save() {
+	_save( manualSave = false ) {
+		if ( this._savePromise ) {
+			if ( manualSave ) {
```
Comment (niegowski) [inline packages/ckeditor5-autosave/src/autosave.js:256]: There is a missing else case coverage.

**Review by niegowski — CHANGES_REQUESTED** [review]
There is a missing CC.

File: packages/ckeditor5-autosave/src/autosave.js (hunk 4 above)
Comment (scofalik) [reply packages/ckeditor5-autosave/src/autosave.js]: Sorry for that, I have problems with CC on my local machine for some reason and had to check it on CI.

**Review by niegowski — APPROVED** [review]

**Pull request merged**